core/jvm/src/Buffer.kt

```diff
@@ -21,557 +21,281 @@
 package kotlinx.io
 
 import kotlinx.io.internal.*
-import java.io.Closeable
 import java.io.EOFException
 import java.io.IOException
 import java.io.InputStream
 import java.io.OutputStream
 import java.nio.ByteBuffer
 import java.nio.channels.ByteChannel
-import java.nio.charset.Charset
-import java.security.InvalidKeyException
-import java.security.MessageDigest
-import javax.crypto.Mac
-import javax.crypto.spec.SecretKeySpec
 
-actual class Buffer : Source, Sink, Cloneable, ByteChannel {
+public actual class Buffer : Source, Sink, Cloneable {
```

Comment:
Since `Buffer` no longer implements `ByteChannel`, is it possible to have a single `common` implementation now?